apps/deposit-pool/src/hooks/useGetHookInfo.ts

```diff
@@ -0,0 +1,194 @@
+  ERC20TransferFromArgs,
+  RPC_URL_MAPPING,
+  TRANSACTION_TYPES,
+  TransactionFactory,
+} from "@bleu/utils/transactionFactory";
+import { useCallback, useMemo } from "react";
+import { Address, maxUint256, parseUnits } from "viem";
+import { BigNumber } from "ethers";
+import { FormType } from "#/types";
+import {
+  AddLiquidity,
+  AddLiquidityKind,
+  AddLiquidityProportionalInput,
+  MAX_UINT256,
+} from "@balancer/sdk";
+
+const addLiquidity = new AddLiquidity();
+
+export function useGetHookInfo(pool?: IPool) {
+  const { cowShedProxy, context } = useIFrameContext();
+  const tokenAllowances = useTokensAllowances({
+    tokenAddresses: pool?.allTokens.map((token) => token.address) || [],
+    spender: cowShedProxy,
+  });
+
+  const defaultPermitData = useMemo(() => {
+    return pool?.allTokens.map((token) => {
+      return {
+        tokenAddress: token.address,
+        amount: maxUint256,
+        tokenSymbol: token.symbol,
+      };
+    });
+  }, [pool]);
+
+  const getPermitData = useCallback(
+    (params: FormType) => {
+      if (!pool) throw new Error("Missing pool");
+
+      if (!tokenAllowances) return defaultPermitData;
+
+      // Check needed allowance with 5% buffer
+      const amountsWithBuffer = pool.allTokens.map((token) => {
+        const tokenAddress = token.address.toLowerCase();
+        const amount = params.amounts[tokenAddress];
+        const amountBigNumber = BigNumber.from(
+          parseUnits(amount.toString(), token.decimals)
+        );
+        return {
+          tokenAddress,
+          amount: amountBigNumber.toBigInt(),
+          tokenSymbol: token.symbol,
+        };
+      });
+
+      // Filter tokens that need approval and build max permit data
+      return (
+        amountsWithBuffer
+          .filter((permitInfo) => {
+            const allowance = tokenAllowances[permitInfo.tokenAddress];
+            return (
+              !allowance || BigNumber.from(allowance).lt(permitInfo.amount)
+            );
+          })
+          .map((permitInfo) => {
+            return {
+              tokenAddress: permitInfo.tokenAddress,
+              amount: maxUint256,
+              tokenSymbol: permitInfo.tokenSymbol,
+            };
+          }) || []
+      );
+    },
+    [pool, tokenAllowances, defaultPermitData]
+  );
+
+  const getPoolDepositTxs = useCallback(
+    async (params: FormType) => {
+      if (!pool || !context || !context.account || !cowShedProxy)
+        throw new Error("Missing context");
+      const poolState = minimalPoolToPoolState(pool);
+
+      const referenceTokenDecimals = pool.allTokens.find(
+        (token) =>
+          token.address.toLowerCase() ===
+          params.referenceTokenAddress.toLowerCase()
+      )?.decimals;
+
+      if (!referenceTokenDecimals) throw new Error("Invalid reference token");
+
+      const addLiquidityInput: AddLiquidityProportionalInput = {
+        chainId: context.chainId,
+        rpcUrl: RPC_URL_MAPPING[context.chainId],
```

Comment:
Couldn't we just use the jsonRpcProvider from useIFrameContext?

Reply:
I think that we can but I think that is easier to use this mapping in this case (it is static anyway).